deepspeed/ops/adagrad/cpu_adagrad.py

```diff
@@ -19,7 +19,7 @@ def __init__(self, model_params, lr=1e-2, eps=1e-10, weight_decay=0, amsgrad=Fal
         self.opt_id = DeepSpeedCPUAdagrad.optimizer_id
         DeepSpeedCPUAdagrad.optimizer_id = DeepSpeedCPUAdagrad.optimizer_id + 1
         self.fp32_optimizer_states = fp32_optimizer_states
-        self.ds_opt_adagrad = CPUAdagradBuilder().load()
+        self.ds_opt_adagrad = CPUAdagradBuilder().set_dtype(self.param_groups[0]['params'][0].dtype).load()
```

Comment:
I am unconvinced that the new `set_dtype` is needed. Could `dtype` not simply be passed into constructor to set `self.dtype`?

Reply:
removed set_dtype function